File: client-app/src/app/models/project.ts
import { Sprint } from './sprint';
import { Assignee } from './assignee';

export interface Project {
    id: string;
    name: string;
    description: string;
    description_text: string;
    priority: string;
    //original_estimated_duration: string;
    //currently_estimated_duration: string;
    //time_logged: string;
    status: string;
    created_at: string;
    sprints: Sprint[];
    assignees: Assignee[];
    //updated_at: string;
    //reporter_id: string;
    //team_id: string;
    //project_id: string;
    //reviewer_id: string;
    //sprint_id: string;
    //assignees: string[];
}

File: Domain/Project.cs
namespace Domain
{
    public class Project
    {
        public Guid Id { get; set; }

        public string team_id { get; set; }

        public string name { get; set; }

        public string description { get; set; }

        public string description_text { get; set; }

        public string priority { get; set; }

        public string owner_id { get; set; }

        public string status { get; set; }

        public DateTime created_at { get; set; }

        public DateTime updated_at { get; set; }

        public ICollection<ProjectSprint> sprints { get; set; } = new List<ProjectSprint>();

        public ICollection<ProjectAssignee> assignees { get; set; } = new List<ProjectAssignee>();
    }
}
